packages/sdk/src/wallet/Ordit.ts

```diff
@@ -233,9 +235,12 @@ export class Ordit {
     return psbt.toHex()
   }
 
-  signMessage(message: string, type?: AddressFormats) {
+  signMessage(message: string, type?: AddressFormats, accountIndex?: number, addressIndex?: number) {
     const addressType = type || this.selectedAddressType
-    const node = this.allAddresses.find((wallet) => wallet.format === addressType) as Account
+    const accountIndexToSign: number = accountIndex === undefined ? 0 : accountIndex;
+    const addressIndexToSign: number = addressIndex === undefined ? 0 : addressIndex;
+    const node = this.allAddresses.find((wallet) => wallet.format === addressType && wallet.derivationPath.account === accountIndexToSign
+      && wallet.derivationPath.addressIndex === addressIndexToSign) as Account
```

Comment:
should we just use `getAddressByType` function here?

Reply:
Wanted to use it initially, but this function has too many conditions. Updated to use this